src/DOMPoint.js

```diff
@@ -21,23 +21,13 @@ const _ = o => {
 export
 class DOMPointReadOnly {
     constructor(x,y,z,w) {
-        if (arguments.length === 1) {
-            if (!isDOMPointInit(x))
-                throw new TypeError('Expected an object with x, y, z, and w properties')
+        _(this).x = (x !== undefined) ? Number(x) : 0
+        _(this).y = (y !== undefined) ? Number(y) : 0
+        _(this).z = (z !== undefined) ? Number(z) : 0
+        _(this).w = (w !== undefined) ? Number(w) : 1
 
-            _(this).x = x.x
-            _(this).y = x.y
-            _(this).z = x.z
-            _(this).w = x.w
-        }
-        else if (arguments.length === 4)  {
-            _(this).x = x || 0
-            _(this).y = y || 0
-            _(this).z = z || 0
-            _(this).w = w || 0
-        }
-        else {
-            throw new TypeError('Expected 1 or 4 arguments')
+        if (this.constructor === DOMPointReadOnly) {
+            Object.freeze(_(this))
```

Comment:
What if someone extends from `DOMPointReadOnly`? The `this.constructor` check may not work as expected?